pkg/core/consensus/initiator/initiator.go

```diff
@@ -0,0 +1,122 @@
+package initiator
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/block"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/factory"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/generation"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/msg"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/consensus/user"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/peer/processing/chainsync"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/encoding"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet"
+	log "github.com/sirupsen/logrus"
+)
+
+var l = log.WithField("process", "consensus initiator")
+
+func LaunchConsensus(eventBroker wire.EventBroker, rpcBus *wire.RPCBus, w *wallet.Wallet, counter *chainsync.Counter) {
```

Comment:
As the task is about `Automate the startup of consensus`, don't we need to call LaunchConsensus at startup? I find it being called only within the CLI commands. (sorry probably I'm misunderstanding something here)

Reply:
It's really quite hard to start consensus if we have no wallet loaded. I think we can also automate that part but I'm not sure if we should take that out of the hands of the user either.

Reply:
Could we load the wallet automatically?

   In case of a panic, the node should be capable of recovering without need of any manual actions like LoadWallet from cli. That said, I assume if wallet.dat and walletDB are existing, node can cope with reloading wallet and restarting the consensus without any manual action. Otherwise if a node is down and then restarted automatically, it will be out of network consensus until the node owner realizes it.

Reply:
That's a very good point, I hadn't considered it. I will move to implement it.

Reply:
Hmm, one thing that does come up is the fact that the node would need to know about the wallet password. Would there be a safe way to keep that somewhere?

Reply:
Indeed.  Dunno safe way. We can decide on this later and just keep it in mind for now.